### PR TITLE
Update code for retrieving changes

### PR DESCRIPTION
**Pull request by davisagli**
image

Call the improved process in CumulusCI for retrieving changes.

Includes an upgrade of cumulusci (and dependencies) and re-adding sfdx to the docker image (it's a dependency of the new retrieval code, but doesn't need to be authenticated to a dev hub).

### CODE REVIEW

**Comment by wlonk** [comment]
@davisagli lgtm, I'll merge it unless you have anything else you want to do/add to it.

File: requirements/dev.txt [hunk 1]
```diff
@@ -10,10 +10,12 @@ attrs==19.3.0             # via black, pytest
 babel==2.7.0              # via sphinx
 backcall==0.1.0           # via ipython
 black==19.10b0
-chardet==3.0.4            # via doc8
+certifi==2019.11.28       # via requests
```
Comment (jgerigmeyer) [inline requirements/dev.txt]: What prompted these changes to requirements?

Comment (davisagli) [reply requirements/dev.txt]: possibly we updated cumulusci in the past without re-running pip-compile to update the pins for its dependencies, or something like that.

Comment (davisagli) [reply requirements/dev.txt]: or maybe we only ran it for prod.txt and forgot dev.txt (which extends prod)

Comment (jgerigmeyer) [reply requirements/dev.txt]: Ok -- I re-locked in `gh-commits` branch.

**Pull request merged**